twenty-first/src/util_types/mmr/mmr_trait.rs

```diff
@@ -1,7 +1,46 @@
+use super::mmr_accumulator::MmrAccumulator;
+use super::mmr_membership_proof::MmrMembershipProof;
 use crate::math::digest::Digest;
 use crate::util_types::algebraic_hasher::AlgebraicHasher;
 
-use super::{mmr_accumulator::MmrAccumulator, mmr_membership_proof::MmrMembershipProof};
+/// A definition of a mutation of a leaf in an MMR
```

Comment:
This documentation comment is non-descriptive. Try explaining, without listing the individual fields or their types, which kind of data this struct encapsulates, and what its instances are for. Avoid the word "mutation" because this word conjures different meanings in the heads of different people. It might also be worth drawing attention to pitfalls or counter-intuitive features -- in this case: the old leaf is not part of this data structure, and its knowledge is not required to apply the mutation.

Reply:
I came up with:
>  A wrapper for the data needed to change the value of a leaf in an MMR when
 only the MMR-accumulator is known, i.e., only the peaks and the leaf-count
are known.